packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -113,8 +114,16 @@ function checkFieldsAgainstIndex(
   if (numFields > keySchema.length && keySchema.length !== 2) {
     throw new InvalidDirectiveError('Too many fields passed in to @connection directive.');
   }
+  // when sort key is passed, ensure that the length of composite sort key matches the length of the fields passed
+  if (numFields > 1) {
+    const querySortFields = inputFieldNames.slice(1);
+    const tableSortFields = String(keySchema[1].AttributeName).split(ModelResourceIDs.ModelCompositeKeySeparator());
+    if (querySortFields.length !== tableSortFields.length) {
+      throw new InvalidDirectiveError(`Invalid @connection directive  ${field.name.value}. fields does not accept partial sort key`);
```

Comment:
I think we could present a better error message here. What length was expected on which field and how many was passed in.